numcodecs/compat.py

```diff
@@ -151,6 +152,12 @@ def ensure_bytes(buf):
         if arr.dtype == object:
             raise TypeError('object arrays are not supported')
 
+        # Force CuPy arrays to NumPy arrays
+        # because they don't have a `tobytes` method (yet)
+        # xref: https://github.com/cupy/cupy/pull/2617
+        if isinstance(arr, cp.ndarray):
+            arr = arr.get()
```

Comment:
As noted, this is a workaround. The real solution here is PR ( https://github.com/cupy/cupy/pull/2617 ), which should land in CuPy 7.0.0. IOW this can probably be ignored for the sake of this discussion.

Reply:
Ooh, +1 to `cupy.ndarray..tobytes`